protocol/contracts/v2/safeGuards/ThresholdSafeGuard.sol

```diff
@@ -0,0 +1,236 @@
+      * Only the Owner is allowed to disable. Never set the owner to the same safe which this is guarding
+      * as it could be bricked.
+     */
+    bool public override disableGuardChecks;
+
+    /**
+     * @notice The required number of signatures required by default for all transactions.
+     */
+    uint256 public override defaultSignaturesThreshold;
+    
+    /**
+      * @notice Per contract and function override of required signature thresholds on the Safe.
+      * If not explicitly set for a contract/function, then the `defaultSignaturesThreshold` is used.
+      * If the threshold is less than the builtin Safe threshold, no signature checks are performed 
+      * in this guard -- the Safe has already verified the signers.
+      */
+    mapping(address => mapping(bytes4 => uint256)) public override functionThresholds;
+
+    /**
+      * @notice Approved addresses which are allowed to execute approved transactions
+      * Safe owners are also allowed to execute.
+     */
+    EnumerableSet.AddressSet internal _safeTxExecutors;
+
+    constructor(address _initialRescuer, address _initialExecutor, uint256 _defaultSignaturesThreshold) 
+        TempleElevatedAccess(_initialRescuer, _initialExecutor)
+    {
+        defaultSignaturesThreshold = _defaultSignaturesThreshold;
+    }
+
+    /**
+     * @notice Disable the Safe Guard, so all transactions pass if the Gnosis threshold of signers is met.
+     *   1. Rescuers setRescueMode(true)
+     *   2. Rescuers setDisableGuardChecks(true)
+     * Note: Rescuers won't have a Guard on their multisig, so this can't be bricked.
+     */
+    function setDisableGuardChecks(bool value) external onlyInRescueMode {
+        disableGuardChecks = value;
+        emit DisableGuardChecksSet(value);
+    }
+
+    /**
+      * @notice Add an address to the allowed list who can execute the transaction
+      * once the minimum threshold of signers have approved
+     */
+    function addSafeTxExecutor(address executor) external onlyElevatedAccess {
+        if (executor == address(0)) revert InvalidAddress();
+        emit SafeTxExecutorAdded(executor);
+        if (!_safeTxExecutors.add(executor)) revert InvalidExecutor();
+    }
+
+    /**
+      * @notice Remove an address from the allowed list who can execute the transaction
+      * once the minimum threshold of signers have approved
+     */
+    function removeSafeTxExecutor(address executor) external onlyElevatedAccess {
+        if (executor == address(0)) revert InvalidAddress();
+        emit SafeTxExecutorRemoved(executor);
+        if (!_safeTxExecutors.remove(executor)) revert InvalidExecutor();
+    }
+
+    /**
+      * @notice Set the default number of signatories required, when a contract/function signature
+      * not explicitly defined.
+     */
+    function setDefaultSignaturesThreshold(uint256 threshold) external onlyElevatedAccess {
+        emit DefaultSignaturesThresholdSet(threshold);
```

Comment:
can we check just to be sure this is not set to 0

Reply:
I'm pretty sure that would be a redundant check. 
* The underlying safe threshold would be >= 1
* If the safe threshold is >= this default threshold then the guard will return regardless. 

So set to 0 or 1 has the same behaviour. Basically means by default things don't need extra signers, but you can explicity set it higher per function